Expensify/App
# [No QA] Remove react-native-community/cli as a hardcoded dependency

### Details
Following up on @kidroca's advice [here](https://github.com/Expensify/App/pull/7117#issuecomment-1011079874), this PR removes `@react-native-community/cli` as an explicit dependency – instead we'll let it be a transitive dependency of RN (properly upgraded now).

This PR also adds `@react-native-community/cli-tools` as an explicit dev dependency because it's used directly in our code.

### Fixed Issues
$ n/a

### Tests
1. Run `npm run clean && npm i && cd ios && pod install && cd ..`
1. Run the app and make sure it runs on all platforms.


### QA Steps
None.

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
